skiplist/src/arena.rs

```diff
@@ -1,76 +1,128 @@
 use std::{
+    cell::Cell,
     mem, ptr,
-    sync::{
-        atomic::{AtomicU32, Ordering},
-        Arc,
-    },
+    sync::atomic::{AtomicUsize, Ordering},
 };
 
+const ADDR_ALIGN_MASK: usize = 7;
+
 struct ArenaCore {
-    len: AtomicU32,
-    cap: usize,
-    ptr: *mut u8,
+    len: AtomicUsize,
+    cap: Cell<usize>,
+    ptr: Cell<*mut u8>,
 }
 
 impl Drop for ArenaCore {
     fn drop(&mut self) {
-        let ptr = self.ptr as *mut u64;
-        let cap = self.cap / 8;
+        let ptr = self.ptr.get() as *mut u64;
+        let cap = self.cap.get() / 8;
         unsafe {
             Vec::from_raw_parts(ptr, 0, cap);
         }
     }
 }
 
+// Thread-local Arena
 pub struct Arena {
-    core: Arc<ArenaCore>,
+    core: ArenaCore,
```

Comment:
As `Arena` is just a direct wrapper of `AreanaCore`, seems no need to introduce `ArenaCore` anymore

Reply:
Done @Connor1996 , PTAL again.